uitools/import/Esri/ArcGISRuntime/Toolkit/FloorFilter.qml

```diff
@@ -0,0 +1,599 @@
+                    Layout.fillWidth: true
+                    orientation: Qt.Horizontal
+                }
+                ToolButton {
+                    id: collapser
+                    objectName: "collapser"
+                    Layout.fillWidth: true
+                    Layout.alignment: Qt.AlignLeft
+                    // use different pointing icon based on the leader position. if icons are collapsed and left position: use right pointing icon; if right position: use left pointing icon; if collapsedIcons === true, switch the icon.
+                    // leaderPosition === right XNOR collapsedIcons; if collapsedIcons === false, expression evaluates opposite as its operands. if collapsedIcons === true, expression evaluates as its operands.
+                    icon.source: !((internal.leaderPosition === FloorFilter.LeaderPosition.UpperRight ||
+                                    internal.leaderPosition === FloorFilter.LeaderPosition.LowerRight) ^ internal.collapsedIcons) ? "images/chevrons-left.svg" : "images/chevrons-right.svg"
+                    text: "Collapse"
+                    flat: true
+                    display: internal.collapsedIcons ? AbstractButton.IconOnly : AbstractButton.TextBesideIcon
+                    onClicked: internal.collapsedIcons = !internal.collapsedIcons
+                }
+                ToolSeparator {
+                    Layout.fillWidth: true
+                    orientation: Qt.Horizontal
+                }
+                ToolButton {
+                    objectName: "zoom"
+                    Layout.fillWidth: true
+                    Layout.alignment: Qt.AlignLeft
+                    icon.source: "images/zoom-to-object.svg"
+                    text: "Zoom to"
+                    flat: true
+                    // enabled based on the currentVisibleListView and if site/facility element is not undefined.
+                    enabled: !listView.visible ? controller.selectedFacilityId : internal.currentVisibileListView === FloorFilter.VisibleListView.Site
+                                                 ? controller.selectedSiteId : controller.selectedFacilityId
+                    display: internal.collapsedIcons ? AbstractButton.IconOnly : AbstractButton.TextBesideIcon
+                    onClicked: {
+                        if (internal.currentVisibileListView === FloorFilter.VisibleListView.Site
+                                && listView.visible)
+                            controller.zoomToSite(controller.selectedSiteId);
+                        else if (internal.currentVisibileListView
+                                 === FloorFilter.VisibleListView.Facility
+                                 || !listView.visible)
+                            controller.zoomToFacility(
+                                        controller.selectedFacilityId);
+                        else
+                            console.error("extra enum not accounted for.");
+                    }
```

Comment:
Removing the checks, the UI takes care of stopping the click. If user _really_ wants to click it, if I remove the checks it will be able to do it by finding the zoom `Button` object and trigger the signal `.clicked(null)`.